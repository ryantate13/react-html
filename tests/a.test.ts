import {a} from '../src';

describe('a', () => {
    it('renders with children', () => {
        expect(a({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(a({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(a()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(a({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(a({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(a({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(a({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(a({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(a({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(a({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(a({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(a({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(a({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(a({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(a({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(a({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(a({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(a({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(a({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(a({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(a({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(a({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(a({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(a({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(a({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(a({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(a({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(a({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(a({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(a({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(a({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(a({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(a({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(a({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(a({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(a({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(a({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(a({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(a({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(a({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(a({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(a({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(a({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(a({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(a({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(a({translate: "no"})).toMatchSnapshot();
    });

    it('renders with download="TEST"', () => {
        expect(a({download: "TEST"})).toMatchSnapshot();
    });

    it('renders with href="https://google.com/"', () => {
        expect(a({href: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with hrefLang="es-419"', () => {
        expect(a({hrefLang: "es-419"})).toMatchSnapshot();
    });

    it('renders with rel="a b c d"', () => {
        expect(a({rel: "a b c d"})).toMatchSnapshot();
    });

    it('renders with target="_blank"', () => {
        expect(a({target: "_blank"})).toMatchSnapshot();
    });

    it('renders with type="application/json"', () => {
        expect(a({type: "application/json"})).toMatchSnapshot();
    });
});