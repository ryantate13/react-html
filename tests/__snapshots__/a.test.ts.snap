// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a renders arrays of children 1`] = `
<a>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</a>
`;

exports[`a renders with accessKey="a,b,c,d" 1`] = `
<a
  accessKey="a,b,c,d"
/>
`;

exports[`a renders with autoCapitalize="characters" 1`] = `
<a
  autoCapitalize="characters"
/>
`;

exports[`a renders with autoCapitalize="none" 1`] = `
<a
  autoCapitalize="none"
/>
`;

exports[`a renders with autoCapitalize="off" 1`] = `
<a
  autoCapitalize="off"
/>
`;

exports[`a renders with autoCapitalize="on" 1`] = `
<a
  autoCapitalize="on"
/>
`;

exports[`a renders with autoCapitalize="sentences" 1`] = `
<a
  autoCapitalize="sentences"
/>
`;

exports[`a renders with autoCapitalize="words" 1`] = `
<a
  autoCapitalize="words"
/>
`;

exports[`a renders with children 1`] = `
<a>
  TEST
</a>
`;

exports[`a renders with className="foo bar baz" 1`] = `
<a
  className="foo bar baz"
/>
`;

exports[`a renders with contentEditable="false" 1`] = `
<a
  contentEditable="false"
/>
`;

exports[`a renders with contentEditable="true" 1`] = `
<a
  contentEditable="true"
/>
`;

exports[`a renders with dir="auto" 1`] = `
<a
  dir="auto"
/>
`;

exports[`a renders with dir="ltr" 1`] = `
<a
  dir="ltr"
/>
`;

exports[`a renders with dir="rtl" 1`] = `
<a
  dir="rtl"
/>
`;

exports[`a renders with download="TEST" 1`] = `
<a
  download="TEST"
/>
`;

exports[`a renders with draggable="false" 1`] = `
<a
  draggable="false"
/>
`;

exports[`a renders with draggable="true" 1`] = `
<a
  draggable="true"
/>
`;

exports[`a renders with empty attributes 1`] = `<a />`;

exports[`a renders with hidden=false 1`] = `
<a
  hidden={false}
/>
`;

exports[`a renders with hidden=true 1`] = `
<a
  hidden={true}
/>
`;

exports[`a renders with href="https://google.com/" 1`] = `
<a
  href="https://google.com/"
/>
`;

exports[`a renders with hrefLang="es-419" 1`] = `
<a
  hrefLang="es-419"
/>
`;

exports[`a renders with id="TEST" 1`] = `
<a
  id="TEST"
/>
`;

exports[`a renders with inputMode="decimal" 1`] = `
<a
  inputMode="decimal"
/>
`;

exports[`a renders with inputMode="email" 1`] = `
<a
  inputMode="email"
/>
`;

exports[`a renders with inputMode="none" 1`] = `
<a
  inputMode="none"
/>
`;

exports[`a renders with inputMode="numeric" 1`] = `
<a
  inputMode="numeric"
/>
`;

exports[`a renders with inputMode="search" 1`] = `
<a
  inputMode="search"
/>
`;

exports[`a renders with inputMode="tel" 1`] = `
<a
  inputMode="tel"
/>
`;

exports[`a renders with inputMode="text" 1`] = `
<a
  inputMode="text"
/>
`;

exports[`a renders with inputMode="url" 1`] = `
<a
  inputMode="url"
/>
`;

exports[`a renders with is="foo" 1`] = `
<a
  is="foo"
/>
`;

exports[`a renders with itemID="https://google.com/" 1`] = `
<a
  itemID="https://google.com/"
/>
`;

exports[`a renders with itemProp="https://google.com/" 1`] = `
<a
  itemProp="https://google.com/"
/>
`;

exports[`a renders with itemRef="foo bar" 1`] = `
<a
  itemRef="foo bar"
/>
`;

exports[`a renders with itemScope=false 1`] = `
<a
  itemScope={false}
/>
`;

exports[`a renders with itemScope=true 1`] = `
<a
  itemScope={true}
/>
`;

exports[`a renders with itemType="https://google.com/" 1`] = `
<a
  itemType="https://google.com/"
/>
`;

exports[`a renders with lang="" 1`] = `
<a
  lang=""
/>
`;

exports[`a renders with lang="es-419" 1`] = `
<a
  lang="es-419"
/>
`;

exports[`a renders with ping="https://google.com/" 1`] = `
<a
  ping="https://google.com/"
/>
`;

exports[`a renders with referrerPolicy="origin" 1`] = `
<a
  referrerPolicy="origin"
/>
`;

exports[`a renders with rel="a b c d" 1`] = `
<a
  rel="a b c d"
/>
`;

exports[`a renders with slot="TEST" 1`] = `
<a
  slot="TEST"
/>
`;

exports[`a renders with spellCheck="false" 1`] = `
<a
  spellCheck="false"
/>
`;

exports[`a renders with spellCheck="true" 1`] = `
<a
  spellCheck="true"
/>
`;

exports[`a renders with style={"color":"#000000"} 1`] = `
<a
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`a renders with tabIndex=1 1`] = `
<a
  tabIndex={1}
/>
`;

exports[`a renders with target="_blank" 1`] = `
<a
  target="_blank"
/>
`;

exports[`a renders with title="TEST" 1`] = `
<a
  title="TEST"
/>
`;

exports[`a renders with translate="no" 1`] = `
<a
  translate="no"
/>
`;

exports[`a renders with translate="yes" 1`] = `
<a
  translate="yes"
/>
`;

exports[`a renders with type="application/json" 1`] = `
<a
  type="application/json"
/>
`;

exports[`a renders without attributes 1`] = `<a />`;
